utils/utils.go

```diff
@@ -205,3 +208,47 @@ func RemoveNewLines(s string) string {
 func DiffToLower(_, old, new string, _ *schema.ResourceData) bool {
 	return strings.EqualFold(old, new)
 }
+
+// ApiResponseInfo - interface over different ApiResponse types from sdks
+type ApiResponseInfo interface {
+	HttpNotFound() bool
+	LogInfo()
+}
+
+// ResourceReadyFunc polls api to see if resource exists based on id
+type ResourceReadyFunc func(ctx context.Context, d *schema.ResourceData) (bool, error)
+
+// WaitForResourceToBeReady - keeps retrying until resource is ready(true is returned), or until err is thrown, or ctx is cancelled
+func WaitForResourceToBeReady(ctx context.Context, d *schema.ResourceData, fn ResourceReadyFunc) error {
```

Comment:
@cristiGuranIonos I noticed that this function is only used for certificate manager. In this PR we also have changes for mongo clusters, users, etc. Did you forget to also use `WaitForResourceToBeReady` for those resources or you just added this as a starting point for future PRs?

Reply:
Exactly, starting point I've implemented it for dataplatform also in the dataplatform PR. The others will do in separate PRs so less code.